rust/src/operations/transaction/state.rs

```diff
@@ -21,16 +23,38 @@ use crate::action::Add;
 use crate::delta_datafusion::{logical_expr_to_physical_expr, to_correct_scalar_value};
 use crate::table_state::DeltaTableState;
 use crate::DeltaResult;
-use crate::{schema, DeltaTableError};
+use crate::DeltaTableError;
 
 impl DeltaTableState {
     /// Get the table schema as an [`ArrowSchemaRef`]
     pub fn arrow_schema(&self) -> DeltaResult<ArrowSchemaRef> {
-        Ok(Arc::new(
-            <ArrowSchema as TryFrom<&schema::Schema>>::try_from(
-                self.schema().ok_or(DeltaTableError::NoMetadata)?,
-            )?,
-        ))
+        let meta = self.current_metadata().ok_or(DeltaTableError::NoMetadata)?;
+        let fields = meta
+            .schema
+            .get_fields()
+            .iter()
+            .filter(|f| !meta.partition_columns.contains(&f.get_name().to_string()))
+            .map(|f| f.try_into())
+            .chain(
+                meta.schema
+                    .get_fields()
+                    .iter()
+                    .filter(|f| meta.partition_columns.contains(&f.get_name().to_string()))
+                    .map(|f| {
+                        let field = ArrowField::try_from(f)?;
+                        let corrected = match field.data_type() {
+                            // TODO Datafusion seems to currently not implement comparsion kernes for booleans in
+                            // dict arrays, so we have to omit converting that type for now.
+                            DataType::Boolean => field.data_type().clone(),
```

Comment:
I think it's actually a good idea to not encode booleans.
```suggestion
                            // Dictionary encoding columns bools is a waste of space
                            DataType::Boolean => field.data_type().clone(),
```